### ISSUE TITLE
False positive: E1136(unsubscriptable-object) in pandas

### ISSUE THREAD

**Issue by sam-s** [PyCQA/pylint#7582]
### Bug description

calling [`dropna`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.dropna.html) causes an `unsubscriptable-object` error:
```a.py
"[E1136(unsubscriptable-object), ] Value 'df' is unsubscriptable"
import pandas as pd

df = pd.DataFrame({"a":[1,2,None]}).dropna()
print(df["a"])
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module pylint
a.py:5: [E1136(unsubscriptable-object), ] Value 'df' is unsubscriptable
```

### Expected behavior

no diagnostics

### Pylint version

```shell
pylint 2.15.3
astroid 2.12.10
Python 3.10.6 (main, Aug 10 2022, 11:40:04) [GCC 11.3.0]
```

### OS / Environment

Linux PopOS

**Comment by sam-s** [comment]
still present with
```
pylint 4.0.5
astroid 4.1.1
Python 3.14.3 (tags/v3.14.3:323c59a, Feb  3 2026, 16:04:56) [MSC v.1944 64 bit (AMD64)]
```
and `pandas=3.0.1`

**Comment by sam-s** [comment]
same with sort_index
